From 4b484f354dac5960d863c5c344b516285909ddd8 Mon Sep 17 00:00:00 2001
From: Ryuinferno <ryuinferno.xda@gmail.com>
Date: Thu, 26 Dec 2013 19:11:23 +0100
Subject: [PATCH 025/201] Camera2: More thorough compatibility fixes

Certain chipsets (such as omap) have contrast, sharpness and saturation
set, but no max integer given for those

Change-Id: I58efaef4d0a511aefc803a440e0c3aa3dbdc9483
---
 src/com/android/camera/CameraSettings.java | 9 ++++++---
 src/com/android/camera/PhotoModule.java    | 9 ++++++---
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index d956eb3ec..fb4ddf758 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -847,15 +847,18 @@ public class CameraSettings {
                     selectableZoneAf, ParametersWrapper.getSupportedSelectableZoneAf(mParameters));
         }
 
-        if (saturation != null && !CameraUtil.isSupported(mParameters, "saturation")) {
+        if (saturation != null && !CameraUtil.isSupported(mParameters, "saturation") &&
+                !CameraUtil.isSupported(mParameters, "saturation-max")) {
             removePreference(group, saturation.getKey());
         }
 
-        if (contrast != null && !CameraUtil.isSupported(mParameters, "contrast")) {
+        if (contrast != null && !CameraUtil.isSupported(mParameters, "contrast") &&
+                !CameraUtil.isSupported(mParameters, "contrast-max")) {
             removePreference(group, contrast.getKey());
         }
 
-        if (sharpness != null && !CameraUtil.isSupported(mParameters, "sharpness")) {
+        if (sharpness != null && !CameraUtil.isSupported(mParameters, "sharpness") &&
+                !CameraUtil.isSupported(mParameters, "sharpness-max")) {
             removePreference(group, sharpness.getKey());
         }
 
diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index d6653153e..09a6e97cf 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -3190,7 +3190,8 @@ public class PhotoModule
         }
 
         //Set Saturation
-        if (CameraUtil.isSupported(mParameters, "saturation")) {
+        if (CameraUtil.isSupported(mParameters, "saturation") &&
+                CameraUtil.isSupported(mParameters, "saturation-max")) {
             String saturationStr = mPreferences.getString(
                     CameraSettings.KEY_SATURATION,
                     mActivity.getString(R.string.pref_camera_saturation_default));
@@ -3201,7 +3202,8 @@ public class PhotoModule
             }
         }
         // Set contrast parameter.
-        if (CameraUtil.isSupported(mParameters, "contrast")) {
+        if (CameraUtil.isSupported(mParameters, "contrast") &&
+                CameraUtil.isSupported(mParameters, "contrast-max")) {
             String contrastStr = mPreferences.getString(
                     CameraSettings.KEY_CONTRAST,
                     mActivity.getString(R.string.pref_camera_contrast_default));
@@ -3212,7 +3214,8 @@ public class PhotoModule
             }
         }
         // Set sharpness parameter
-        if (CameraUtil.isSupported(mParameters, "sharpness")) {
+        if (CameraUtil.isSupported(mParameters, "sharpness") &&
+                CameraUtil.isSupported(mParameters, "sharpness-max")) {
             String sharpnessStr = mPreferences.getString(
                     CameraSettings.KEY_SHARPNESS,
                     mActivity.getString(R.string.pref_camera_sharpness_default));
-- 
2.17.1

